amps/ags/rm-community/rm-community-repo/unit-test/java/org/alfresco/module/org_alfresco_module_rm/disposition/RetentionScheduleModelUnitTest.java

```diff
@@ -0,0 +1,121 @@
+        RetentionSchedule result = apiNodesModelFactory.mapRetentionScheduleData(dispositionSchedule);
+
+        assertEquals(dispositionSchedule.getNodeRef().getId(),result.getId());
+        assertEquals(dispositionSchedule.getDispositionAuthority(),result.getAuthority());
+        assertEquals(dispositionSchedule.getDispositionInstructions(),result.getInstructions());
+        assertEquals(dispositionSchedule.isRecordLevelDisposition(),result.isRecordLevel());
+    }
+
+    @Test
+    public void mapRetentionScheduleActionDefDataTest()
+    {
+        // Mock data
+        DispositionActionDefinition dispositionActionDefinition = mock(DispositionActionDefinition.class);
+        NodeRef nodeRef = mock(NodeRef.class);
+        Period mockPeriod = mock(Period.class);
+        QName mockQName = mock(QName.createQName("qname"));
+        when(dispositionActionDefinition.getNodeRef()).thenReturn(nodeRef);
+        when(dispositionActionDefinition.getName()).thenReturn(RETAIN_STEP);
+        when(dispositionActionDefinition.getDescription()).thenReturn("Description");
+        when(dispositionActionDefinition.getIndex()).thenReturn(1);
+        when(dispositionActionDefinition.getGhostOnDestroy()).thenReturn("ghost");
+        when(dispositionActionDefinition.getPeriod()).thenReturn(mockPeriod);
+        when(dispositionActionDefinition.getPeriodProperty()).thenReturn(mockQName);
+        when(dispositionActionDefinition.getLocation()).thenReturn("location");
+        // Call the method
+        RetentionScheduleActionDefinition result = apiNodesModelFactory.mapRetentionScheduleActionDefData(dispositionActionDefinition);
+        String period = result.getPeriod()+"|"+result.getPeriodAmount();
+        // Assertions
+        assertEquals(dispositionActionDefinition.getNodeRef().getId(),result.getId());
+        assertEquals(dispositionActionDefinition.getName(),result.getName());
+        assertEquals(dispositionActionDefinition.getDescription(),result.getDescription());
+        assertEquals(dispositionActionDefinition.getIndex(),result.getIndex());
+        assertEquals(dispositionActionDefinition.getLocation(),result.getLocation());
+        assertEquals(dispositionActionDefinition.getPeriodProperty().getLocalName(),result.getPeriodProperty());
+        assertEquals(dispositionActionDefinition.getPeriod(),new Period(period));
+    }
+
+    @Test
+    public void mapRetentionScheduleOptionalInfoTest()
+    {
+        // Mock data
+        RetentionSchedule retentionSchedule = new RetentionSchedule();
+        DispositionSchedule schedule = mock(DispositionSchedule.class);
+        List<String> includeParam = List.of("actions");
+        when(schedule.getDispositionActionDefinitions()).thenReturn(List.of(mock(DispositionActionDefinition.class)));
+        // Call the method
+        apiNodesModelFactory.mapRetentionScheduleOptionalInfo(retentionSchedule, schedule, includeParam);
+        // Assertions
+        assertNotNull(retentionSchedule);
```

Comment:
I don't think this is what you should be asserting here. The assertion should be about the list of actions.

Reply:
We have removed the test as it is not required.